cf/api/services_test.go

```diff
@@ -193,26 +195,54 @@ var _ = Describe("Services Repo", func() {
 	})
 
 	Describe("creating a service instance", func() {
-		It("makes the right request with the async flag", func() {
+		It("makes the right request", func() {
 			setupTestServer(testapi.NewCloudControllerTestRequest(testnet.TestRequest{
 				Method:   "POST",
 				Path:     "/v2/service_instances?accepts_incomplete=true",
-				Matcher:  testnet.RequestBodyMatcher(`{"name":"instance-name","service_plan_guid":"plan-guid","space_guid":"my-space-guid","async":true}`),
+				Matcher:  testnet.RequestBodyMatcher(`{"name":"instance-name","service_plan_guid":"plan-guid","space_guid":"my-space-guid"}`),
 				Response: testnet.TestResponse{Status: http.StatusCreated},
 			}))
 
-			err := repo.CreateServiceInstance("instance-name", "plan-guid")
+			err := repo.CreateServiceInstance("instance-name", "plan-guid", nil)
 			Expect(testHandler).To(HaveAllRequestsCalled())
 			Expect(err).NotTo(HaveOccurred())
 		})
 
+		Context("when there are parameters", func() {
+			It("sends the parameters as part of the request body", func() {
+				setupTestServer(testapi.NewCloudControllerTestRequest(testnet.TestRequest{
+					Method:   "POST",
+					Path:     "/v2/service_instances?accepts_incomplete=true",
+					Matcher:  testnet.RequestBodyMatcher(`{"name":"instance-name","service_plan_guid":"plan-guid","space_guid":"my-space-guid","parameters": {"data": "hello"}}`),
+					Response: testnet.TestResponse{Status: http.StatusCreated},
+				}))
+
+				paramsMap := make(map[string]interface{})
+				paramsMap["data"] = "hello"
+
+				err := repo.CreateServiceInstance("instance-name", "plan-guid", paramsMap)
+				Expect(testHandler).To(HaveAllRequestsCalled())
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("and there is a failure during serialization", func() {
```

Comment:
Notice that the value here is `paramsMap`, a map instead of a string. The user's input actually gets unmarshalled earlier, so if the json is invalid, it will raise an error there. By the time we get to this part of the code, we know for a fact that the object is not "almost" valid json. Look at https://github.com/cloudfoundry/cli/pull/441/files#diff-e39828f6ca182457b7f7a8842b138547R110